Objects/rangeobject.c

```diff
@@ -1135,13 +1145,19 @@ range_iter(PyObject *seq)
 
     /* If all three fields and the length convert to long, use the int
      * version */
-    lstart = PyLong_AsLong(r->start);
-    if (lstart == -1 && PyErr_Occurred()) {
+    lstop = PyLong_AsLong(r->stop);
+    if (lstop == -1 && PyErr_Occurred()) {
         PyErr_Clear();
         goto long_range;
     }
-    lstop = PyLong_AsLong(r->stop);
-    if (lstop == -1 && PyErr_Occurred()) {
+
+    if (r->start == _PyLong_GetZero() && r->step == _PyLong_GetOne() ) {
```

Comment:
How large is overhead of two `PyLong_AsLong()` calls for 0 and 1? Is it noticeable?

Run something like:
```
./python -m timeit -s 'r = range(10)' 'for i in r: break'
```
AFAIK it is the fastest code that is affected by the iterator creation time.

Reply:
The overhead for the two `PyLong_AsLong` calls (plus some bits and pieces) are from the first benchmark above:
```
range(1): Mean +- std dev: [main_pgo] 44.4 ns +- 1.0 ns -> [pr_pgo] 34.7 ns +- 4.0 ns: 1.28x faster
```
So about 10 ns.

The last benchmark from the first comment:
```
for loop: Mean +- std dev: [main_pgo] 295 ns +- 9 ns -> [pr_pgo] 267 ns +- 5 ns: 1.10x faster
```
executes:
```
def g():
    x=0
    for ii in range(10):
        x += 1
```
So a small for loop with minimal work becomes 10% faster. (timings may vary, my system is noisy)